quinn/scala_to_pyspark.py

```diff
@@ -46,7 +46,7 @@ def display(self):
         print("".join(self.lines()))
 
     def clean_function_definition(self, s):
-        m = re.search('(\s*)def (\w+).*\((.*)\)', s)
+        m = re.search(r"(\s*)def (\w+).*\((.*)\)", s)
```

Comment:
```suggestion
        m = re.search(r"(\s*)def (\w+).*?\((.*)\)", s)
```

The fix looks good to me. You might want the search for first bracket to be non-greedy. The below example is for Python but I am not sure if this is used for Python or Scala function definition parsing. Definitions with brackets inside the parameters might cause this problem.

```
>>> s.clean_function_definition("""def foo(a, b, c=(1))""") # greedy current version
'def foo(1)):\n'
>>> s.clean_function_definition("""def foo(a, b, c=(1))""") # non-greedy
'def foo(a, b, c=(1)):\n'
```